service/worker/workerdeployment/version_workflow.go

```diff
@@ -728,4 +699,107 @@ func (d *VersionWorkflowRunner) updateVersionStatusAfterDrainageStatusChange(new
 		// This should only happen if we encounter an error while checking the drainage status of the version
 		d.VersionState.Status = enumspb.WORKER_DEPLOYMENT_VERSION_STATUS_UNSPECIFIED
 	}
+
+	v := workflow.GetVersion(ctx, "Step1", workflow.DefaultVersion, 0)
+	if v != workflow.DefaultVersion {
+		err := d.syncVersionStatusToTaskQueues(ctx)
+		if err != nil {
+			d.logger.Error("failed to sync version status after drainage status change", "error", err)
+		}
+	}
+
+}
+
+// syncVersionStatusToTaskQueues syncs the current version status to all task queues.
+// This function acquires the workflow lock the same way as the update handlers do so that
+// there are no race conditions during task-queue sync. It is called internally when the status
+// of the version becomes draining/drained.
```

Comment:
```suggestion
// of the version becomes draining or drained.
```
Do you also call it when the version becomes active after it was previously draining or drained?

Reply:
not really but I can see where the confusion lies; this function was developed with the sole purpose of being called *after* the drainage go-routine changes the version status of a version to draining/drained. All other version changes (such as changing the version from draining -> current/ramping) flow nicely through the function `syncState`, which in turn calls the helper `syncVersionDataToTaskQueues`. 

With that being said, anyone can have the question you raised and I suspect this is because of the way I named my function. I changed it so that one can clearly understand why and when it's called.

Reply:
ok, so just confirming, when a version becomes active, the task queue will change the status to active in it's VersionInfo also?